pkg/connector/client.go

```diff
@@ -182,8 +211,16 @@ func (wa *WhatsAppClient) Connect(ctx context.Context) {
 	if err := wa.Main.updateProxy(ctx, wa.Client, false); err != nil {
 		zerolog.Ctx(ctx).Err(err).Msg("Failed to update proxy")
 	}
-	wa.startLoops()
-	if err := wa.Client.Connect(); err != nil {
+	startCtx := wa.startLoops()
+
+	if err := wa.Client.Connect(); exerrors.IsNetworkError(err) && wa.Main.Config.MaxConnectRetries != 0 {
```

Comment:
Is all this new code necessary?

It's weird to have this retry loop here, but we also have events.Connected and events.Disconnected in handleWAEvent.

Feels like we should refactor so that Connect() just kicks the bridge into the world where it tries to stay connected and let the events in handleWAEvent update the bridge state like they already do.

Reply:
Hmm yeah, there could be a new flag in whatsmeow.Client to allow autoreconnect on connect, then Connect would just do

```go
go cli.dispatchEvent(&events.Disconnected{})
go cli.autoReconnect()
```

before returning an error (same as it does for disconnections)

whatsmeow probably needs some more context plumbing too

Reply:
I like this pattern, there's no reason a network error on first connect should be special compared to a network error once we are running and get disconnected.

This is a pattern we should follow in all bridges